cli/internal/runcache/runcache.go

```diff
@@ -155,8 +153,9 @@ func (tc *TaskCache) RestoreOutputs(ctx context.Context, prefixedUI *cli.Prefixe
 		}
 	} else {
 		// If no outputs have changed, that means we have a local cache hit.
-		cacheStatus.Local = true
-		prefixedUI.Warn(fmt.Sprintf("Skipping cache check for %v, outputs have not changed since previous run.", tc.pt.TaskID))
+		// We will still check to see whether the cache exists, so we can get the duration.
+		cacheStatus = tc.rc.cache.Exists(tc.hash)
```

Comment:
This is the purpose of this PR. One thing that's weird here is before we were _assuming_ that `hasChangedOutputs == false` means cache hit. In order to get the timeSaved, we have to check the cache. That means:

1. the thing we end up returning is no longer an assumption, we're actually checking the existence of cache
2. the assumption can be wrong / different from what the actual cache status is (e.g. if the `hasChangedOutputs` implementation is wrong)

I considered doing this instead:

```suggestion
		cacheStatus = newFSTaskCacheStatus(true, 0)
		cacheStatus.TimeSaved = tc.rc.cache.Exists(tc.hash).TimeSaved
```

but I feel like that's an equally weird implementation, since we're combining two sources of truth.

Open to suggestions!

Reply:
Can we store the time saved with the daemon alongside the globs? Then you either get "no change, here's time saved", check fs cache, or check http cache

Reply:
This is done now